### DISCUSSION TITLE
Wezterm types

### DISCUSSION THREAD

**Discussion by andresatierf** [folke/lazydev.nvim#9] [Q&A]
Could this be used to enable wezterm types when editing wezterm configuration? or just all the time?

**Comment by folke** [comment]
Yes of course, just follow what's in the readme?

```lua
  { "justinsgithub/wezterm-types", lazy = true },
  {
    "folke/lazydev.nvim",
     ....
    opts = {
      library = {
        wezterm = vim.env.LAZY .. "/wezterm-types",
      },
    },
  },
```

you may also have to force `opts.enabled = true`, since your weszterm config wont be in your neovim config folder

**Reply by andresatierf** [reply]
I tried that but it didn't work, hence my question. I use LazyVim and tried to emulate what it's done there but with no luck. I tried several iterations of the same thing:
- `vim.env.LAZY .. "/wezterm-types"`
- `LazyVim.get_plugin_path("wezterm-types")`
- `LazyVim.get_plugin_path("wezterm-types", "types")`
- `"wezterm-types/types"`

![image](https://github.com/folke/lazydev.nvim/assets/29847057/e183a111-121a-4e5b-89fe-b14821a1b1fd)

![image](https://github.com/folke/lazydev.nvim/assets/29847057/30c4dabf-35e0-43c6-8719-72dc545376ee)

I also tried it with neodev, before updating lazyvim, using the `.neoconf.json` in the wezterm directory but to no avail.
![image](https://github.com/folke/lazydev.nvim/assets/29847057/0703a8f0-92c5-44f3-99e1-f9f84fb21fc5)

**Reply by folke** [reply]
```lua
library = {
...
wezterm = "wezterm-types"
...
}
```

**Reply by folke** [reply]
That `vim.env.LAZY` or `LAyVim.get_plugin_path` is no longer needed

**Reply by andresatierf** [reply]
Still no luck. I checked in `:LspInfo` and the root path looks alright

**Reply by mgastonportillo** [reply]
> Still no luck. I checked in `:LspInfo` and the root path looks alright

I think you're just using the wrong types, right? Try `Wezterm` and `Config` instead of `wezterm` and `WeztermConfig`

**Reply by andresatierf** [reply]
You're right, I took that from a different wezterm types repo and when I changed to `justinsgithub`'s I didn't check if the types were called the same

**Comment by mgastonportillo** [comment]
Here's a working example:
```lua
  { "Bilal2453/luvit-meta" },
  { "justinsgithub/wezterm-types" },
  {
    "folke/lazydev.nvim",
    ft = "lua",
    opts = {
      enabled = true,
      library = {
        "luvit-meta/library",
        "wezterm-types/types",
        vim.fn.stdpath "data" .. "/lazy/ui/nvchad_types",
        --[[
          vim.fn.expand "$HOME/workspace/neovim/ui/nvchad_types",
          vim.fn.stdpath "data" .. "/lazy/lazy.nvim/lua/lazy",
          vim.fn.expand "$VIMRUNTIME/lua/vim",
          "${3rd}/luv/library",
        ]]
      },
    },
  },
```
![Screenshot 2024-06-03 132548](https://github.com/folke/lazydev.nvim/assets/106234166/18bc3d63-0c4a-4e85-8ca8-e1b7c681775a)
